packages/firestore/test/integration/api/get_options.test.ts

```diff
@@ -418,15 +418,37 @@ apiDescribe('GetOptions', persistence => {
 
   it('get non existing doc while offline with default get options', () => {
     return withTestDocAndInitialData(persistence, null, docRef => {
-      return docRef.firestore
-        .disableNetwork()
+      return (
+        docRef.firestore
+          .disableNetwork()
+          // Attempt to get doc. This will fail since there's nothing in cache.
+          .then(() => docRef.get())
+          .then(
+            doc => {
+              expect.fail();
+            },
+            expected => {}
+          )
+      );
+    });
+  });
+
+  // TODO(b/112267729): We should raise a fromCache=true event with a
+  // nonexistent snapshot, but because the default source goes through a normal
+  // listener, we do not.
+  // tslint:disable-next-line:ban skipping test on purpose.
+  it.skip('get deleted doc while offline with default get options', () => {
+    return withTestDocAndInitialData(persistence, null, docRef => {
+      return docRef
+        .delete()
+        .then(() => docRef.firestore.disableNetwork())
         .then(() => docRef.get())
-        .then(
-          doc => {
-            expect.fail();
-          },
-          expected => {}
-        );
+        .then(doc => {
+          expect(doc.exists).to.be.false;
+          expect(doc.data).to.be.null;
```

Comment:
This should be doc.data() and `undefined`.

Reply:
Wow, good catch. Now I know I can just send you my tests to evaluate rather than actually run them.